server/data/data-ldap/src/test/java/org/apache/james/user/ldap/LdapHealthCheckTest.java

```diff
@@ -0,0 +1,41 @@
+package org.apache.james.user.ldap;
+
+import org.apache.james.core.healthcheck.Result;
+import org.apache.james.domainlist.api.mock.SimpleDomainList;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class LdapHealthCheckTest {
+
+    LdapHealthCheck ldapHealthCheck;
+    ReadOnlyUsersLDAPRepository ldapUserRepository;
+    LdapGenericContainer ldapContainer;
+
+    @BeforeEach
+    public void setUp() {
+        ldapUserRepository = new ReadOnlyUsersLDAPRepository(new SimpleDomainList());
+        ldapHealthCheck = new LdapHealthCheck(ldapUserRepository);
+        ldapContainer = DockerLdapSingleton.ldapContainer;
+    }
+
+    @Test
+    void checkShouldReturnUnhealthyIfLdapIsDown() {
+        ldapContainer.pause();
+
+        Result checkResult = ldapHealthCheck.check().block();
+        assertNotNull(checkResult);
+        assertTrue(checkResult.isUnHealthy());
```

Comment:
Use assertj library:

```suggestion
        assertThat(checkResult.isUnHealthy()).isTrue();
```

This is a common project practice and it tends to be more readable than Junit assertions.